### DISCUSSION TITLE
Queuing with router.routeMessageByChannelId

### DISCUSSION THREAD

**Discussion by elavy-harris** [nextgenhealthcare/connect#5650] [Q&A] [answered]
I'm using `router.routeMessageByChannelId` to send a message to another channel. If that other channel is not deployed (especially during startup/shutdown), the function will fail. Is there any way to queue such messages? I could make the calling channel dependent on the receiving channel, but that feels rather heavy-handed.

Any other suggestions?

**Comment by pacmano1** [comment, marked as answer]
Actually, your heavy handed approach is correct IMHO.   It's a simply dependency problem.   

You can of course use a JS writer which can queue.

**Reply by elavy-harris** [reply]
Thank you.

I don't think the JS Writer idea fits. A Channel Writer can queue, too. The whole point is that I'm not using `router.routeMessageByChannelId ` as part of a destination.
